temporalio/worker/worker.py

```diff
@@ -279,14 +317,9 @@ async def shutdown(self) -> None:
         bridge_shutdown_task = asyncio.create_task(self._bridge_worker.shutdown())
         # Wait for the poller loops to stop
         await self._task
-        # Shutdown the workers
-        worker_tasks: List[asyncio.Task] = []
+        # Shutdown the activity worker (there is no workflow worker shutdown)
         if self._activity_worker:
-            worker_tasks.append(
-                asyncio.create_task(self._activity_worker.shutdown(graceful_timeout))
-            )
-        # TODO(cretz): Support workflow worker
-        await asyncio.wait(worker_tasks)
+            await self._activity_worker.shutdown(graceful_timeout)
```

Comment:
Does `_activity_worker.run` complete without being requested to shutdown?

Reply:
Nope, I got the explanation I was looking for (worker shutsdown when it sees ShutdownError)